### PR TITLE
[AArch64][SME] Implement inline-asm clobbers for za/zt0

### PR DESCRIPTION
**Pull request by MDevereau**
This enables specifing "za" or "zt0" to the clobber list for inline asm. This complies with the acle SME addition to the asm extension here:
https://github.com/ARM-software/acle/pull/276

### CODE REVIEW

File: llvm/lib/Target/AArch64/AArch64ISelLowering.cpp [hunk 1]
```diff
@@ -10702,6 +10702,14 @@ AArch64TargetLowering::getRegForInlineAsmConstraint(
       parseConstraintCode(Constraint) != AArch64CC::Invalid)
     return std::make_pair(unsigned(AArch64::NZCV), &AArch64::CCRRegClass);
 
+  if (StringRef("{za}").equals_insensitive(Constraint)) {
```
Comment (sdesmalen-arm) [inline llvm/lib/Target/AArch64/AArch64ISelLowering.cpp:10705]:
```suggestion
  if (Constraint == "za") {
```
?

File: llvm/lib/Target/AArch64/AArch64RegisterInfo.cpp [hunk 2]
```diff
@@ -507,6 +507,10 @@ bool AArch64RegisterInfo::isAsmClobberable(const MachineFunction &MF,
         MCRegisterInfo::regsOverlap(PhysReg, AArch64::X16))
     return true;
 
+  // ZA/ZT0 registers are reserved but may be permitted in the clobber list.
+  if (PhysReg.id() == AArch64::ZA || PhysReg.id() == AArch64::ZT0)
```
Comment (sdesmalen-arm) [inline llvm/lib/Target/AArch64/AArch64RegisterInfo.cpp:511]:
```suggestion
  if (PhysReg == AArch64::ZA || PhysReg == AArch64::ZT0)
```

File: llvm/lib/Target/AArch64/AArch64ISelLowering.cpp [hunk 3]
```diff
@@ -10702,6 +10702,14 @@ AArch64TargetLowering::getRegForInlineAsmConstraint(
...
+    return std::make_pair(unsigned(AArch64::ZA), &AArch64::MPRRegClass);
+  }
+
+  if (StringRef("{zt0}").equals_insensitive(Constraint)) {
```
Comment (sdesmalen-arm) [inline llvm/lib/Target/AArch64/AArch64ISelLowering.cpp:10709]:
```suggestion
  if (Constraint == "zt0") {
```
?

Comment (MDevereau) [reply llvm/lib/Target/AArch64/AArch64ISelLowering.cpp:10709]: Done

**Review by sdesmalen-arm — APPROVED** [review]
LGTM!

**Pull request merged**